app/src/test/java/org/mozilla/fenix/home/RecentTabsListFeatureTest.kt

```diff
@@ -361,6 +376,181 @@ class RecentTabsListFeatureTest {
         homeStore.waitUntilIdle()
 
         assertEquals(1, homeStore.state.recentTabs.size)
-        assertEquals(selectedTab, homeStore.state.recentTabs[0])
+        assert(homeStore.state.recentTabs[0] is RecentTab.Tab)
+        assertEquals(selectedTab, (homeStore.state.recentTabs[0] as RecentTab.Tab).tabSessionState)
+    }
+
+    @Test
+    fun `GIVEN a selected tab group WHEN the feature starts THEN dispatch the selected tab group as a recent tab list`() {
+        val tab = createTab(
+            url = "https://www.mozilla.org",
+            id = "1",
+            historyMetadata = HistoryMetadataKey(
+                url = "https://www.mozilla.org",
+                searchTerm = "test search term",
+                referrerUrl = "https://www.mozilla.org"
+            )
+        )
+        val tabs = listOf(tab)
+        val browserStore = BrowserStore(
+            BrowserState(
+                tabs = tabs,
+                selectedTabId = "1"
+            )
+        )
+        val feature = RecentTabsListFeature(
+            browserStore = browserStore,
+            homeStore = homeStore
+        )
+
+        feature.start()
+
+        homeStore.waitUntilIdle()
+
+        assertEquals(2, homeStore.state.recentTabs.size)
+        assert(homeStore.state.recentTabs[0] is RecentTab.Tab)
+        assert(homeStore.state.recentTabs[1] is RecentTab.SearchGroup)
```

Comment:
nit: Somewhat duplicate because you assert that it is a group on the line below. 🙂

Reply:
True,  I'll remove the first check.